### DISCUSSION TITLE
Custom Maven repository

### DISCUSSION THREAD

**Discussion by redstonedesigner** [renovatebot/renovate#11573] [Request Help] [answered]
Hi there,

Just starting to deploy Renovate within my team's self-hosted Gitea instance.  One thing I'm struggling with is configuring repository credentials for the Renovate runner.  Ideally, I'd like to declare this in the config file that I use to run Renovate, however I cannot work out how to do that.

Current configuration:

```
GITHUB_COM_TOKEN=ghp_[REDACTED]
RENOVATE_PLATFORM=gitea
RENOVATE_AUTODISCOVER=true
RENOVATE_BASE_DIR=/tmp/renovate/
LOG_LEVEL=info
RENOVATE_TOKEN=[REDACTED]
RENOVATE_ENDPOINT=https://git.example.com/api/v1
RENOVATE_GIT_AUTHOR=Renovate <renovate@[REDACTED]>
RENOVATE_USEEMOJI=false
```

Running Renovate using `docker run --rm --env-file /home/user/renovate/.env renovate/renovate`

EDIT:  Sorry, I realised I created some confusion.  The platform credentials are fine.  I'm struggling to configure a private Maven repository that Renovate can access and check.

**Comment by viceice** [comment, marked as answer]
?? Aleways use `LOG_LEVEL=debug` when start working. We need some debug logs to help you.

**Reply by redstonedesigner** [reply]
I feel like I might not have been clear, and just came here to edit and upate my message.

The repository credentials are working fine.  I need to configure a private Maven repository for my team's dependencies.  The repository is located at `https://nexus.example.com/repository/project_name` and requires authentication to access.

**Reply by viceice** [reply]
So you need a host rule

https://docs.renovatebot.com/getting-started/running/#global-config

**Reply by redstonedesigner** [reply]
I agree that the host rule feature seems to be the best.  How do I configure it to be a Maven host & provide the URL, username & password required?

**Reply by viceice** [reply]
https://docs.renovatebot.com/getting-started/private-packages/

**Reply by redstonedesigner** [reply]
Using DEBUG logs:

`DEBUG: Adding password authentication for https://maven.exaple.com/repository/project/ to hostRules`
...
`Failed to look up dependency org.project:dependency`

I've verified that the account has access to the repositories and all packages within.

EDIT: Updated configuration value:
`RENOVATE_CONFIG={"hostRules":[{"matchHost":"https://maven.example.com/repository/project/","password":"[REDACTED]","username":"renovate"}]}`

**Reply by viceice** [reply]
You need a package rule to assign your custom registry url.

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#registryurls

**Reply by redstonedesigner** [reply]
The URL is declared in the `pom.xml` file and access is attempted.

**Reply by viceice** [reply]
Please provide debug logs

**Reply by redstonedesigner** [reply]
The error appeared to be that Nexus OSS doesn't automatically rebuild the `maven-metadata.xml` files when an artifact is updated.  I've just added it as a task so hopefully this problem will not occur again.
